packages/x-data-grid-premium/src/components/GridAggregationHeader.tsx

```diff
@@ -1,93 +1,78 @@
 import * as React from 'react';
-import composeClasses from '@mui/utils/composeClasses';
-import { styled } from '@mui/material/styles';
+import clsx from 'clsx'
+import { css, useStyled } from '@mui/x-data-grid/internals';
 import {
-  getDataGridUtilityClass,
-  gridClasses,
-  GridColDef,
   GridColumnHeaderParams,
   GridColumnHeaderTitle,
 } from '@mui/x-data-grid';
 import type { GridBaseColDef } from '@mui/x-data-grid/internals';
 import { getAggregationFunctionLabel } from '../hooks/features/aggregation/gridAggregationUtils';
 import { useGridApiContext } from '../hooks/utils/useGridApiContext';
 import { useGridRootProps } from '../hooks/utils/useGridRootProps';
-import { DataGridPremiumProcessedProps } from '../models/dataGridPremiumProps';
 
-interface OwnerState extends DataGridPremiumProcessedProps {
-  classes: DataGridPremiumProcessedProps['classes'];
-  colDef: GridColDef;
-}
-
-const GridAggregationHeaderRoot = styled('div', {
+const headerStyled = css({
   name: 'MuiDataGrid',
-  slot: 'AggregationColumnHeader',
-  overridesResolver: (_, styles) => styles.aggregationColumnHeader,
-})<{ ownerState: OwnerState }>({
-  display: 'flex',
-  flexDirection: 'column',
-  [`&.${gridClasses['aggregationColumnHeader--alignRight']}`]: {
+  slot: 'aggregationColumnHeader',
+}, {
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  alignRight: {
     alignItems: 'flex-end',
   },
-  [`&.${gridClasses['aggregationColumnHeader--alignCenter']}`]: {
```

Comment:
Without this PR, you would have probably migrated to use the `variant` API at some point, is that a DX your proposal could support?

Reply:
I don't think we have a need for it in the grid, but yes, the props are passed to `useStyled` so we could add variant classnames to the `.root` one.

Reply:
It's mostly a question of coherence in the DX, `[`&.${gridClasses['aggregationColumnHeader--alignCenter']}`]: {` and `variants: { props: {  alignCenter: true }}` are doing exactly the same thing (from what I can understand of this code).

And since the pickers and tree view are using the variant API extensively, the new way of authoring CSS should support it :ok_hand: